### ISSUE TITLE
Unable to update challenge :: Response does not complete challenge

### ISSUE THREAD

**Issue by b9chris** [ebekker/ACMESharp#131]
This either isn't a bug, or isn't a bug yet. Right now I'm just looking for The Right Way To Do this:

1. New-ACMEIdentifier
2. Complete-ACMEChallenge
3. Submit-ACMEChallenge

Did this for 3 domains; the first 2 went straight to valid. The third went to invalid. Now that I've arrived at invalid, how do I start over?

When I run New-ACMEIdentifier, I'm told the key is already present and I can't proceed. The key I believe is the Alias. I'm reusing the Domain as the Alias to keep things simple going forward as I'm planning on using all of these in a single SAN cert I update over time - having a complex map of Aliases is not desired.

I assume this is the root of my problem. Can I delete the "key" (alias) and start over properly?

Proceeding to Complete-ACMEChallenge given New-ACMEIdentifier won't fire, I get a successful response that says Status : pending. This would suggest I'm back on track, and on the right track.

I then run Submit-ACMEChallenge and I get the error in the title, back off track.

So, I think I just don't know the right way to do this. What would that be?

**Comment by bseddon** [comment]
You can only use an alias once.  If you want to start over or create an identifier for a different domain you must use a different alias.  Aliases are just a string of characters that mean something to you.  They are a short hand for a collection of information generated by the Lets Encrypt servers.

If you try to use an alias for another domain it will be reported as invalid.

**Comment by b9chris** [comment]
I see now. Definitely no bug then, but I do wonder if "Alias" is a bad name choice. Alias implies something friendly to me that I can use in the future. The truth it appears is this is a key that can be used at most for one attempt to get Let's Encrypt to approve a domain, regardless of that outcome - it may be throwaway if it ends in invalid. To me that should be named something more to its purpose like:

RequestId
CaseId
ValidationKey

I particularly like CaseId as it makes it clear you're talking about this series of commands to get a domain validated but nothing more nothing less.

Request is iffy because it implies a request like perhaps a nonce, only good for one command. ValidationKey is wordy and while more relevant in wording less clear in what it means.

**Comment by bseddon** [comment]
Technically, the parameter is not called 'alias'.  This is just name I used in the text of my response.  The parameter is called 'IdentifierRef', 'Identifier' or just 'Ref' (not case sensitive).

I used the word alias because an identifier is a named entity that represents an otherwise long and unmemorable string generated by ACMESharp to interact with the LE servers in regard to a specific domain validation instance.  In this context, 'alias' works for me. You will use the same IdentifierRef with all commands relating to the same information held by the LE servers potentially far into the future.

The IdentifierRef is not invalid.  It may reference a collection of data on the LE servers that represent an invalidated domain challenge but the reference remains and will remain a valid reference to that information for 294 days.  Why 294?  Who knows.  The LE API designers probably had a good reason.

If you feel strongly about a specific name for this (or any other) parameter and you are comfortable writing C# then adding a synonym for a parameter is a simple change to make,  PowerShell command parameters can be decorated with an 'Alias' (really 'alias' this time) attribute.  So you would find the parameter to which you'd like to add a synonym and add an attribute:
```
[Alias('Whatever')]
```
You would make the change to all 'ACMEIdentifier' command and submit your suggestion as a pull request to be incorporated in the main code base. 

There is a time that an identifier is referenced by the term 'alias'.  This is when you want to process all the identifiers you have created over time. Suppose you want to retrieve the identifier instance that you named 'abc'.  You would do so using something like:

```
Get-ACMEIdentifier | Where-Object{ $_.Alias -eq 'abc' }
```

In this case, 'Alias' is a fixed name as it is the name of a property on an anonymous class created to return selected values from the identifier.

**Comment by reidca** [comment]
I completely agree with @b9chris on this issue. I had exactly the same problem and found the use of the parameter name "Alias" confusing. In the other cmdlets this is known as "IdentifierRef" so I wonder why it is also not called that in  New-ACMEIdentifier.

It would be useful to update the quick start with information about the fact that this identifier cannot be reused if it has gone to a failed state.
